### PR TITLE
Concurrency sweep early backoff

### PR DESCRIPTION
**Pull request by nv-braf**
Add early backoff functionality to quick search w/ concurrency sweep

### CODE REVIEW

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

File: model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py [hunk 1]
```diff
@@ -80,6 +81,9 @@ def set_last_results(
             self, measurements: List[Optional[RunConfigMeasurement]]) -> None:
         self._rcg.set_last_results(measurements)
 
+    def get_last_results(self) -> Optional[RunConfigMeasurement]:
```
[the remark this answers is not in the document]
Comment (nv-braf) [reply model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py:84]: That didn't seem very clean to me. I thought it better to just keep the measurement stored once in the lower level generator and then add an accessor function to re-acquire it when needed.

[the remark this answers is not in the document]
Comment (tgerdesnv) [reply model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py:84]: The results are already passed to this object via set_last_results. That is the clean place to grab/keep them if you need them.  Just append them to a local list, and clear it at the start of each concurrency sweep.

Instead what is happening is the results are being passed through inactive(!) generators RCG->MCG->PACG, and then you are creating a new API to get the data from that low level object.

[the remark this answers is not in the document]
Comment (nv-braf) [reply model_analyzer/config/generate/quick_plus_concurrency_sweep_run_config_generator.py]: I've removed get_last_results(). Please re-review

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 2]
```diff
@@ -77,8 +74,8 @@ def __init__(self,
         self._generator_started = False
 
         self._last_results: List[RunConfigMeasurement] = []
-        self._concurrency_results: List[RunConfigMeasurement] = []
-        self._batch_size_results: List[RunConfigMeasurement] = []
+        self._concurrency_results: List[Optional[RunConfigMeasurement]] = []
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:77]: Is this still correct? I don't think it is

File: model_analyzer/config/generate/perf_analyzer_config_generator.py [hunk 3]
```diff
@@ -181,8 +206,11 @@ def _step(self) -> None:
             self._step_batch_size()
 
     def _add_best_throughput_to_batch_sizes(self) -> None:
-        if self._concurrency_results:
-            best = max(self._concurrency_results)
+        if self._concurrency_results and not any(
+                x is None for x in self._concurrency_results):
+            # type is List[Optional[RCM]], but static checking doesn't understand
```
Comment (tgerdesnv) [inline model_analyzer/config/generate/perf_analyzer_config_generator.py:211]: is this still correct/needed?

**Review by tgerdesnv — CHANGES_REQUESTED** [review]

File: model_analyzer/config/generate/perf_analyzer_config_generator.py (hunk 3 above)
Comment (nv-braf) [reply model_analyzer/config/generate/perf_analyzer_config_generator.py]: Yes

File: model_analyzer/config/generate/perf_analyzer_config_generator.py (hunk 2 above)
Comment (nv-braf) [reply model_analyzer/config/generate/perf_analyzer_config_generator.py:77]: Ye

**Review by tgerdesnv — DISMISSED** [review]

**Pull request merged**